File: update_fonts.py
import os
import re

def update_typography(filepath):
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()
    
    orig = content
    
    # Remove font-display
    content = re.sub(r'\bfont-display\s*', '', content)
    content = re.sub(r'\s*font-display\b', '', content)
    
    # Downsize texts
    content = re.sub(r'\btext-\[80px\]\b', 'text-5xl', content)
    content = re.sub(r'\btext-\[56px\]\b', 'text-4xl', content)
    content = re.sub(r'\btext-8xl\b', 'text-5xl', content)
    content = re.sub(r'\btext-7xl\b', 'text-5xl', content)
    content = re.sub(r'\btext-6xl\b', 'text-4xl', content)
    content = re.sub(r'\btext-5xl\b', 'text-4xl', content)
    
    # cleanup double spaces in classNames
    content = content.replace('  ', ' ')
    
    if orig != content:
        with open(filepath, 'w', encoding='utf-8') as f:
            f.write(content)
        print(f"Updated {filepath}")

dirs = ['components', 'app']
for d in dirs:
    for root, _, files in os.walk(d):
        for file in files:
            if file.endswith('.tsx'):
                update_typography(os.path.join(root, file))

File: update_fonts.js
const fs = require('fs');
const path = require('path');

function updateTypography(filepath) {
    const orig = fs.readFileSync(filepath, 'utf8');
    let content = orig;
    
    // Remove font-display
    content = content.replace(/\bfont-display\s*/g, '');
    content = content.replace(/\s*font-display\b/g, '');
    
    // Downsize texts
    content = content.replace(/\btext-\[80px\]\b/g, 'text-5xl');
    content = content.replace(/\btext-\[56px\]\b/g, 'text-4xl');
    content = content.replace(/\btext-8xl\b/g, 'text-5xl');
    content = content.replace(/\btext-7xl\b/g, 'text-5xl');
    content = content.replace(/\btext-6xl\b/g, 'text-4xl');
    content = content.replace(/\btext-5xl\b/g, 'text-4xl');
    
    // cleanup double spaces
    content = content.replace(/  /g, ' ');
    
    if (orig !== content) {
        fs.writeFileSync(filepath, content, 'utf8');
        console.log(`Updated ${filepath}`);
    }
}

function walkDir(dir) {
    const files = fs.readdirSync(dir);
    for (const file of files) {
        const fullPath = path.join(dir, file);
        if (fs.statSync(fullPath).isDirectory()) {
            walkDir(fullPath);
        } else if (fullPath.endsWith('.tsx')) {
            updateTypography(fullPath);
        }
    }
}

['components', 'app'].forEach(d => walkDir(d));
